docs/book/driver-deployment/upgrade.md

```diff
@@ -7,9 +7,57 @@ Before upgrading driver, please make sure to keep roles and privileges up to dat
+The workaround for this issue is to remount vSAN file services volumes used by the application pods at the same location from the Node VM Guest OS directly.
+
+Find all RWM volumes on the Cluster
+
+```bash
+# kubectl get pv -o wide | grep RWX
+pvc-7e43d1d3-2675-438d-958d-41315f97f42e   1Gi        RWX            Delete           Bound    default/www-web-0   file-sc                 107m   Filesystem
+```
+
+Find all Nodes to which RWM volume is attached. In here below example, we can see volume is attached and mounted on the `k8s-worker3` node.
+
+```bash
+# kubectl get volumeattachment | grep pvc-7e43d1d3-2675-438d-958d-41315f97f42e
+csi-3afe670705e55e0679cba3f013c78ff9603333fdae6566745ea5f0cb9d621b20   csi.vsphere.vmware.com   pvc-7e43d1d3-2675-438d-958d-41315f97f42e   k8s-worker3   true       22s
+```
+
+Log in to the `k8s-worker3` node VM and execute the following command to know where the volume is mounted to.
+
+```bash
+root@k8s-worker3:~# mount | grep pvc-7e43d1d3-2675-438d-958d-41315f97f42e
+10.83.28.38:/52d7e15c-d282-3bae-f64d-8851ad9d352c on /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount type nfs4 (rw,relatime,vers=4.1,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=10.244.3.134,local_lock=none,addr=10.83.28.38)
+```
+
+unmount and remount volume at the same location with the same mount options used for mounting volume.
+
+```bash
+root@k8s-worker3:~# umount -l /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount
+root@k8s-worker3:~# mount -t nfs4 -o sec=sys,minorversion=1  10.83.28.38:/52d7e15c-d282-3bae-f64d-8851ad9d352c /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount
+```
+
+confirm mount point is accessible from the Node VM
+
+```bash
+root@k8s-worker3:~# mount | grep pvc-7e43d1d3-2675-438d-958d-41315f97f42e
+10.83.28.38:/52d7e15c-d282-3bae-f64d-8851ad9d352c on /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount type nfs4 (rw,relatime,vers=4.1,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=10.83.26.244,local_lock=none,addr=10.83.28.38)
+
+root@k8s-worker3:~# ls -la /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount
+total 4
+drwxrwxrwx 3 root root    0 Aug  9 16:48 .
+drwxr-x--- 3 root root 4096 Aug  9 18:40 ..
+-rw-r--r-- 1 root root    6 Aug  9 16:48 test
+```
+
+After all vSAN file share volumes are remounted on the worker VMs, you can proceed with upgrading the driver.
+
+We are in the process of automating the above steps so until we have automation available we have to execute the workaround step mentioned above manually.
```

Comment:
I'm curious to know what is the automation we are mentioning here. 

So, after upgrading the CSI driver to the latest, why not ask the customer to delete the pod and let Kubernetes recreate it. Of course, I'm assuming that they have a Deployment or StatefulSet managing the pod here and there is no standalone Pod.

Reply:
> So, after upgrading the CSI driver to the latest, why not ask the customer to delete the pod and let Kubernetes recreate it.

If we delete the pod, pod will be recreated with the same old driver which does not hostnetwork set to true.

and for frozen volumes, when we delete the pod, the pod remains in the terminating state. That's the reason we are providing a workaround of remounting volume from Guest VM.


> I'm curious to know what is the automation we are mentioning here.

we are thinking about providing a container image to fix all such volumes to help remount on the node VM. In addition to automation, it will help eliminate the requirement of installing `nfs-common` utils on worker VMs.